kernel/kernel-api/src/main/java/io/delta/kernel/internal/types/DataTypeJsonSerDe.java

```diff
@@ -131,22 +132,29 @@ public static StructType deserializeStructType(String structTypeJson) {
    *   "metadata" : { }
    * }
    * </pre>
+   *
+   * @param fieldPath Path from the nearest ancestor that is of the {@link StructField} type.
+   * @param collationMap Maps the path of a {@link StringType} to its collation. Only maps non-UTF8_BINARY collated {@link StringType}.
    */
-  static DataType parseDataType(JsonNode json) {
+  static DataType parseDataType(
+      JsonNode json, String fieldPath, HashMap<String, String> collationMap) {
```

Comment:
you can just use `Map` in arguments def instead of `HashMap`. Same in other places.